### PR TITLE
8331027: JDK's ct.sym file packs corrupted module-info classes

### PR DESCRIPTION
**Pull request by lahodaj**
When writing the `ModuleMainClass` attribute, `CreateSymbols` uses an UTF8 entry (incorrect), instead of a Class_info (correct). This patch fixes that.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331027](https://bugs.openjdk.org/browse/JDK-8331027): JDK's ct.sym file packs corrupted module-info classes (**Bug** - P4)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**) ⚠️ Review applies to [bde7257c](https://git.openjdk.org/jdk/pull/19031/files/bde7257ccbf23f7c4bb5a04aa7f90449bc5ba333)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19031/head:pull/19031` \
`$ git checkout pull/19031`

Update a local copy of the PR: \
`$ git checkout pull/19031` \
`$ git pull https://git.openjdk.org/jdk.git pull/19031/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19031`

View PR using the GUI difftool: \
`$ git pr show -t 19031`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19031.diff">https://git.openjdk.org/jdk/pull/19031.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19031#issuecomment-2088312349)

### CODE REVIEW

File: test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTestImpl.java [hunk 1]
```diff
@@ -1048,7 +1055,114 @@ protected boolean includeEffectiveAccess(ClassList classes, ClassDescription cla
...
+
+
+        Path ctSym = output.resolve("ct.sym");
+
+        deleteRecursively(ctSym);
+
+        CreateSymbols.ALLOW_NON_EXISTING_CLASSES = true;
+        CreateSymbols.EXTENSION = ".class";
+
+        deleteRecursively(ctSym);
```
Comment (asotona) [inline test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTestImpl.java:1104]: Is double deletion needed for some kind of race condition safety?

**Review by asotona — APPROVED** [review]
Looks good to me.

File: test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTestImpl.java (hunk 1 above)
Comment (lahodaj) [reply test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTestImpl.java]: I don't think this is needed. Thanks for noticing, removed.

**Pull request closed without merging**